xarray/core/computation.py

```diff
@@ -445,7 +446,7 @@ def apply_dict_of_variables_vfunc(
     for name, variable_args in zip(names, grouped_by_name):
         core_dim_present = _check_core_dims(signature, variable_args, name)
         if core_dim_present is True:
-            result_vars[name] = func(*variable_args)
+            result_vars[name] = func(*variable_args, keep_attrs=keep_attrs)
```

Comment:
I think that can be any function and there is no guarantee that it will consume this argument (admittedly I am surprised why this does not error in our test suite - so I may be wrong). Also isn't this on the `DataArray` level while the missing attrs are on the `Dataset` level?

Reply:
Yes, good chance that we are undertesting here. But the error is raised for the attribute on `z1` (`DataArray` level). And this propagates `keep_attrs`.

Reply:
OK this only works for `core_dim_present` @mathause, so the attributes would also have to be removed on `on_missing_core_dim == "copy"`. But as @mathause already mentioned, it's debatable if it should loose it's attrs.